src/analysis/retail/shaman/elemental/modules/talents/Stormkeeper.tsx

```diff
@@ -46,6 +287,241 @@ class Stormkeeper extends Analyzer {
     this.damageDoneByBuffedCasts += event.amount + (event.absorbed || 0);
   }
 
+  _explainTimelineWithDetails(cast: SKCast) {
+    const checklistItem = {
+      performance: cast.timeline.performance,
+      summary: <span>Spell order</span>,
+      details: <span>Spell order: See below</span>,
+      check: 'stormkeeper-timeline',
+      timestamp: cast.event.timestamp,
+    };
+
+    const showPrecastSop = SPELLS_SOP_BUFF_REQUIRED.includes(cast.firstRotationCast.ability.guid);
+    const showPrecastMote = [this.stSpender.id].includes(cast.firstRotationCast.ability.guid);
+
+    const showPrecastAny = showPrecastSop || showPrecastMote;
+
+    const extraDetails = (
+      <div
+        style={{
+          overflowX: 'scroll',
+        }}
+      >
+        {showPrecastAny && (
+          <p style={{ marginBottom: '0px' }}>
+            <strong>Precast</strong>
+          </p>
+        )}
+        {showPrecastSop && (
+          <p>
+            <SpellIcon spell={TALENTS.SURGE_OF_POWER_TALENT} />:{' '}
+            <PerformanceMark
+              perf={
+                cast.firstRotationCast.meta?.isEnhancedCast
+                  ? QualitativePerformance.Good
+                  : QualitativePerformance.Fail
+              }
+            />
+          </p>
+        )}
+
+        {showPrecastMote && (
+          <p>
+            <SpellIcon spell={TALENTS.MASTER_OF_THE_ELEMENTS_TALENT} />:{' '}
+            <PerformanceMark
+              perf={
+                cast.firstRotationCast.meta?.isEnhancedCast
+                  ? QualitativePerformance.Good
+                  : QualitativePerformance.Fail
+              }
+            />
+          </p>
+        )}
+        <EmbeddedTimelineContainer
+          secondWidth={60}
+          secondsShown={(cast.timeline.end - cast.timeline.start) / 1000}
+        >
+          <SpellTimeline>
+            <Casts
+              start={cast.event.timestamp}
+              movement={undefined}
+              secondWidth={60}
+              events={cast.timeline.events}
+            />
+          </SpellTimeline>
+        </EmbeddedTimelineContainer>
+      </div>
+    );
+
+    return { extraDetails, checklistItem };
+  }
+
+  /**
+   * Determine the performance of how much maelstrom the user had when casting SK
+   */
+  _determineMaelstromPerformance(maelstromRequired: number, cast: SKCast): QualitativePerformance {
```

Comment:
these underscore-prefixed methods should probably be marked `private`

Reply:
Ah. I come from python where we don't have such luxury. 

Fixed as suggested.